nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -238,10 +238,37 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
       collision_checker_->costAtPose(pose.pose.position.x, pose.pose.position.y), transformed_plan,
       linear_vel, x_vel_sign);
 
+    if (cancelling_) {
+      const double & dt = control_duration_;
+      auto linear_delta = speed.linear.x - linear_vel;
```

Comment:
I'm not totally grokking this logic. Why not find the `speed.linear.x - dt * params_->cancel_decelleration` (or `+` if reversing)?

I'm not sure why you need to compute `speed.linear.x - linear_vel` first at all and the later max/min feels like that is always quashed to the `dt  * decel` value

Reply:
The original logic I had was taking into account that the RPPC could already break and it would make sure to break at least that hard. For example when slowing down at the end of paths or during cusp points. I

I've implemented you suggestion to simplify the code.